### PR TITLE
[AppConfig] Add snapshot support

### PR DESCRIPTION
**Pull request by YalinLi0312**
Resolves: https://github.com/Azure/azure-sdk-for-python/issues/27032

This PR added snapshot operations including:
```python
begin_create_snapshot()
archive_snapshot()
recover_snapshot()
get_snapshot()
list_snapshot()
list_snapshot_configuration_settings()
```

### CODE REVIEW

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 1]
```diff
@@ -501,7 +510,6 @@ def set_read_only(
         :type read_only: bool
         :keyword match_condition: The match condition to use upon the etag
         :paramtype match_condition: :class:`~azure.core.MatchConditions`
-        :keyword str etag: check if the ConfigurationSetting is changed. Set None to skip checking etag
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:504]: Why was this removed?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 2]
```diff
@@ -559,6 +567,195 @@ def set_read_only(
...
+    @distributed_trace
+    def archive_snapshot(
+        self,
+        name: str,
+        *,
+        match_condition: MatchConditions = MatchConditions.Unconditionally,
+        etag: Optional[str] = None,
+        **kwargs
+    ) -> ConfigurationSettingSnapshot:
+        """Archive a configuration setting snapshot. It will update the status of a snapshot from "ready" to "archived".
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:626]: We could add more details here regarding what it means for the snapshot to be archived. For example, that after being archived, it will remain recoverable for the set `retention_period`, after which it will be deleted.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py (hunk 1 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:504]: Because we are using `configuration_setting.etag` instead of `etag` keyword to check "if_match" and "if_non_match".

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/__init__.py [hunk 3]
```diff
@@ -16,16 +16,19 @@
...
 )
+from ._generated.models import Snapshot, ConfigurationSettingFilter
 from ._version import VERSION
 from ._azure_appconfiguration_error import ResourceReadOnlyError
 
 __version__ = VERSION
 __all__ = [
     "AzureAppConfigurationClient",
     "ConfigurationSetting",
+    "ConfigurationSettingFilter",
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/__init__.py:27]: Should they be under azure.appconfiguration or azure.appconfiguration.models?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py (hunk 1 above)
Comment (xiangyan99) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:504]: I don't understand. Can you give more details?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 # Licensed under the MIT License.
 # ------------------------------------
 import json
-from typing import Any, Union, List, Dict, Optional
+from typing import Any, Dict, List, Optional, Union
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:6]: Seems like just re-ordering? :)

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py [hunk 5]
```diff
@@ -91,6 +93,7 @@ def __init__(self, base_url: str, credential: AsyncTokenCredential, **kwargs) ->
             authentication_policy=credential_policy,
             user_agent_policy=user_agent_policy,
             per_call_policies=self._sync_token_policy,
+            api_version="1.0",
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py:96]: Do we allow users to specify api_version?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py [hunk 6]
```diff
@@ -126,16 +129,16 @@ def list_configuration_settings(
...
-         used as wildcard in the beginning or end of the filter
+            used as wildcard in the beginning or end of the filter
         :type key_filter: str
         :param label_filter: filter results based on their label. '*' can be
-         used as wildcard in the beginning or end of the filter
+            used as wildcard in the beginning or end of the filter
         :type label_filter: str
         :keyword datetime accept_datetime: filter out ConfigurationSetting created after this datetime
-        :keyword List[str] fields: specify which fields to include in the results. Leave None to include all fields
+        :keyword list[str] fields: specify which fields to include in the results. Leave None to include all fields
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py:141]: I think it should be typing.List here?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/__init__.py (hunk 3 above)
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/__init__.py:27]: Why we want to expose `ConfigurationSettingFilter`?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py [hunk 7]
```diff
@@ -4,17 +4,18 @@
 # license information.
 # -------------------------------------------------------------------------
 import binascii
-from typing import Any, Optional, Mapping, Union
+from typing import Any, Dict, List, Mapping, Optional, Union
 from requests.structures import CaseInsensitiveDict
```
Comment (kashifkhan) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py:8]: is there a particular reason to use CaseInsensitiveDict from requests ? There is one in azure core itself that is similar.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 8]
```diff
@@ -4,7 +4,7 @@
 # license information.
 # -------------------------------------------------------------------------
 import binascii
-from typing import Optional, Any, Union, Mapping
+from typing import Any, Dict, List, Mapping, Optional, Union
 from requests.structures import CaseInsensitiveDict
```
Comment (kashifkhan) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:8]: same here about case insensitive dict...should use the one from azure core

Comment (annatisch) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:8]: Yes this needs to be removed

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 9]
```diff
@@ -85,7 +88,7 @@ def __init__(self, base_url: str, credential: TokenCredential, **kwargs) -> None
             )
 
         self._impl = AzureAppConfiguration(
-            base_url, pipeline=pipeline, credential_scopes=self._credential_scopes
+            base_url, pipeline=pipeline, credential_scopes=self._credential_scopes, api_version="1.0"
```
[the remark this answers is not in the document]
Comment (annatisch) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:91]: Why is the api version being hard-coded?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py (hunk 1 above)
Comment (annatisch) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:504]: It looks like this was a bug - or at least, we were documenting behaviour that we weren't actually doing.
Though etag should be supported as an optional parameter as well. This would overwrite whatever is in the object.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/__init__.py (hunk 3 above)
Comment (annatisch) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/__init__.py:27]: Yeah - doesn't look like theres a need for this to be public.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 10]
```diff
@@ -559,6 +562,248 @@ def set_read_only(
...
+        :param name: The name of the snapshot to create.
+        :type name: str
+        :param filters: A list of filters used to filter the configuration settings by key field and label field
+            included in the snapshot. Each filter should be in format {"key": <value>, "label": <value>},
+            and key field value is required.
+        :type filters: list[dict[str, str]]
+        :keyword str composition_type: The composition type describes how the key-values within the
+            snapshot are composed. The 'all' composition type includes all key-values. The 'group_by_key'
+            composition type ensures there are no two key-values containing the same key. Known values are:
+            "all" and "group_by_key".
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:587]: Can we type this as `Literal['all', 'group_by_key']`

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 11]
```diff
@@ -559,6 +562,248 @@ def set_read_only(
...
+                **kwargs
+            )
+        except HttpResponseError as error:
+            e = error_map[error.status_code]
+            raise e(message=error.message, response=error.response)
+        except binascii.Error:
+            raise binascii.Error("Connection string secret has incorrect padding")
+
+    @distributed_trace
+    def list_snapshot_configuration_settings(self, name: str, **kwargs) -> ItemPaged[ConfigurationSetting]:
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:775]: While it's not blocking to this PR, I still think I would advocate we expose this API as an overload to the existing `list_configuration_settings` method rather than a separate method.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py (hunk 5 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py]: Yes, "api_version" is a keyword. Now changed to set the default value to "1.0".

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py (hunk 4 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:6]: Yup, just have reordering updates in this file. I reverted the custom models for snapshot.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py (hunk 7 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py]: Updated to use the one from azure-core.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py (hunk 8 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py]: Same as async, has updated.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/__init__.py (hunk 3 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/__init__.py:27]: `Snapshot` has an attribute in type `ConfigurationSettingFilter`. Since we make the generated model `Snapshot` to be public, will also need to make its attributes' non-primary types to be public.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py (hunk 10 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:571]: Updated

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py]: Have changed to get the value from kwargs, and use "1.0" as the default one.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py (hunk 1 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py]: Added the line back and track the bug in this issue: https://github.com/Azure/azure-sdk-for-python/issues/30528

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py (hunk 6 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py:145]: It is in lowercase `list` which means the built-in type used in code instead of typing.List.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 12]
```diff
@@ -56,7 +64,7 @@ class AzureAppConfigurationClient:
     """
 
     # pylint:disable=protected-access
-    def __init__(self, base_url: str, credential: TokenCredential, **kwargs) -> None:
+    def __init__(self, base_url: str, credential: TokenCredential, *, api_version: str = "1.0", **kwargs) -> None:
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:67]: Do we want to show this kwarg in intellisense?

File: sdk/appconfiguration/azure-appconfiguration/CHANGELOG.md [hunk 13]
```diff
@@ -10,6 +10,8 @@
 - Fixed async `update_sync_token` to use async/await keywords
 
 ### Other Changes
+- Set the default `api_version` to "1.0".
```
Comment (kashifkhan) [inline sdk/appconfiguration/azure-appconfiguration/CHANGELOG.md:13]: since there are new operations added they should be under new features :)

Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/CHANGELOG.md:13]: Ah, you're right, I missed that.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py (hunk 12 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:67]: Why not? 1. more clear to show customers that `api_version` is configurable in api  2. checked other languages, e.g. in .Net, [`api_version`](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClientOptions.cs#L45) is showing up in their apiview

Comment (xiangyan99) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:67]: Other Python SDKs do not explicitly show this kwarg.

File: sdk/appconfiguration/azure-appconfiguration/CHANGELOG.md (hunk 13 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/CHANGELOG.md:14]: Updated!

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py (hunk 12 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:67]: ACR does([link](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py#L68)), I remember it was requested to add. I can check whether to expose it here with v-team.

Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:67]: Based on the api [guideline](https://azure.github.io/azure-sdk/general_design.html#service-api-versions): 
"The purposes of the client library is to communicate with an Azure service. Azure services support multiple API versions. To understand the capabilities of the service, the client library **must** be able to support multiple service API versions."
I think it is required to support multiple api versions. @xiangyan99 Do you know why other packages choose not to?

Comment (xiangyan99) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:67]: Took offline.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/__init__.py [hunk 14]
```diff
@@ -16,16 +16,19 @@
...
     "ResourceReadOnlyError",
     "FeatureFlagConfigurationSetting",
     "SecretReferenceConfigurationSetting",
+    "Snapshot",
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/__init__.py:31]: @annatisch it looks to me `Snapshot` is a return-only type and users don't need to create a `Snapshot` object. In this case, do we need to expose `Snapshot`?

Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/__init__.py:31]: Btw, my understanding is that we should make all custom types - including input types and output types - in api surface to be public,

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py (hunk 12 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py]: Reverted the surface updates here, and target the latest `api_version` to `2022-11-01-preview` as it include snapshot operations.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py (hunk 5 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py]: Have reverted the updates.

**Review by kashifkhan — APPROVED** [review]

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 15]
```diff
@@ -133,6 +144,7 @@ def _create_appconfig_pipeline(self, credential, base_url=None, aad_mode=False,
                 DistributedTracingPolicy(**kwargs),
                 HttpLoggingPolicy(**kwargs),
                 ContentDecodePolicy(**kwargs),
+                RequestIdPolicy(**kwargs),
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:147]: I don't think `RequestIdPolicy` should be this far down the pipeline - pretty sure this should be before the retry policy - can you check the default generated pipeline to confirm?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 16]
```diff
@@ -541,6 +553,182 @@ def set_read_only(
...
+            ensures there are no two key-values containing the same key. The 'key_label' composition type ensures
+            there are no two key-values containing the same key and label.
+        :keyword int retention_period: The amount of time, in seconds, that a snapshot will remain in the
+            archived state before expiring. This property is only writable during the creation of a
+            snapshot. If not specified, will set to 2592000(30 days). If specified, should be
+            in range 3600(1 hour) to 7776000(90 days).
+        :keyword dict[str, str] tags: The tags of the snapshot.
+        :return: A poller for create snapshot operation. Call `result()` on this object to wait for the
+            operation to complete and get the created snapshot.
+        :rtype: :class:`~azure.core.polling.LROPoller[~azure.appconfiguration.Snapshot]`
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:586]: We shouldn't use the `:class:` tag within `:rtype:`
```suggestion
        :rtype: ~azure.core.polling.LROPoller[~azure.appconfiguration.Snapshot]
```

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 17]
```diff
@@ -541,6 +553,182 @@ def set_read_only(
...
+        etag: Optional[str] = None,
+        **kwargs
+    ) -> Snapshot:
+        """Archive a configuration setting snapshot. It will update the status of a snapshot from "ready" to "archived".
+        The retention period will start to count, the snapshot will expire when the entire retention period elapses.
+
+        :param name: The name of the configuration setting snapshot to archive.
+        :type name: str
+        :keyword match_condition: The match condition to use upon the etag.
+        :type match_condition: :class:`~azure.core.MatchConditions`
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:613]: We shouldn't use `:class:` in a `:type:` or `:paramtype:` tag either.
```suggestion
        :type match_condition: ~azure.core.MatchConditions
```

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 18]
```diff
@@ -541,6 +553,182 @@ def set_read_only(
...
+            raise binascii.Error("Connection string secret has incorrect padding")
+
+    @distributed_trace
+    def list_snapshot_configuration_settings(self, name: str, **kwargs) -> ItemPaged[ConfigurationSetting]:
+        """List the configuration settings stored under a snapshot in the configuration service, optionally filtered by
+        accept_datetime.
+
+        :param str name: The snapshot name.
+        :keyword datetime accept_datetime: Filter out ConfigurationSetting created after this datetime
+        :keyword list[str] fields: Specify which fields to include in the results. Leave None to include all fields
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:714]: These keyword args should be in the method signature with keyword-only syntax and typing

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 19]
```diff
@@ -3,9 +3,17 @@
...
+
+from azure.core.rest import HttpResponse
 from ._generated._serialization import Model
-from ._generated.models import KeyValue
+from ._generated.models import KeyValue, Snapshot as GeneratedSnapshot, ConfigurationSettingFilter
+
+if sys.version_info >= (3, 8):
+    from typing import Literal  # pylint: disable=no-name-in-module, ungrouped-imports
+else:
+    from typing_extensions import Literal  # type: ignore  # pylint: disable=ungrouped-imports
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:16]: We don't really need this if/else section.
You can just use:
`from typing_extensions import Literal` - and this will always work and it does this check for you.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 20]
```diff
@@ -351,3 +357,114 @@ def _to_generated(self) -> KeyValue:
...
+    """
+    
+    def __init__(
+        self,
+        *,
+        filters: List[Dict[str, str]],
+        composition_type: Optional[Literal["key", "key_label"]] = None,
+        retention_period: Optional[int] = None,
+        tags: Optional[Dict[str, str]] = None,
+        ** kwargs
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:407]:
```suggestion
        **kwargs
```

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 21]
```diff
@@ -351,3 +357,114 @@ def _to_generated(self) -> KeyValue:
...
+         snapshot. If not specified, the default lifetime of key-value revisions will be used.
+        :paramtype retention_period: int
+        :keyword tags: The tags of the snapshot.
+        :paramtype tags: dict[str, str]
+        """
+        self.name = kwargs.get("name", None)
+        self.status = kwargs.get("status", None)
+        self.filters = filters
+        self.composition_type = composition_type
+        self.created = kwargs.get("created", None)
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:429]: Why are we setting all the readonly properties from kwargs?
I'm not sure this is the best way to do this - if it's only to support the `_from_generated` method, we can set them outside the constructor.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py [hunk 22]
```diff
@@ -79,8 +87,7 @@ def __init__(self, base_url: str, credential: AsyncTokenCredential, **kwargs) ->
                 )
             else:
                 raise TypeError(
-                    "Please provide an instance from azure-identity "
-                    "or a class that implement the 'get_token protocol"
+                    "Please provide an instance from azure-identity or a class that implement the 'get_token' protocol"
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py:90]:
```suggestion
                    "Please provide an instance from azure-identity or a class that implements the 'get_token' protocol"
```

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py [hunk 23]
```diff
@@ -539,6 +546,182 @@ async def set_read_only(
...
+            raise binascii.Error("Connection string secret has incorrect padding")
+
+    @distributed_trace
+    def list_snapshot_configuration_settings(self, name: str, **kwargs) -> AsyncItemPaged[ConfigurationSetting]:
+        """List the configuration settings stored under a snapshot in the configuration service, optionally filtered by
+        accept_datetime.
+
+        :param str name: The snapshot name.
+        :keyword datetime accept_datetime: Filter out ConfigurationSetting created after this datetime
+        :keyword list[str] fields: Specify which fields to include in the results. Leave None to include all fields
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py:707]: Keyword-only syntax.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py (hunk 15 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:147]: It is on top of the stack in the default pipeline, I'll change the sequence here.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py (hunk 19 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:16]: No, it'll have import issue without this check. Or do you mean it adds the check now?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 24]
```diff
@@ -545,6 +574,187 @@ def set_read_only(
...
+
+        :param name: The name of the configuration setting snapshot to retrieve.
+        :type name: str
+        :keyword list[str] fields: Specify which fields to include in the results. Leave None to include all fields.
+        :return: The Snapshot returned from the service.
+        :rtype: ~azure.appconfiguration.Snapshot
+        """
+        try:
+            generated_snapshot = self._impl.get_snapshot(
+                name=name, if_match=None, if_none_match=None, select=fields, **kwargs
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:696]: Why aren't we exposing the match_condition + etag here?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 25]
```diff
@@ -545,6 +574,187 @@ def set_read_only(
...
+        self, name: str, *, accept_datetime: Optional[str] = None, fields: Optional[List[str]] = None, **kwargs
+    ) -> ItemPaged[ConfigurationSetting]:
+        """List the configuration settings stored under a snapshot in the configuration service, optionally filtered by
+        accept_datetime and fields to present in return.
+
+        :param str name: The snapshot name.
+        :keyword str accept_datetime: Filter out ConfigurationSetting created after this datetime
+        :keyword list[str] fields: Specify which fields to include in the results. Leave None to include all fields
+        :return: An iterator of :class:`~azure.appconfiguration.ConfigurationSetting`
+        :rtype: ~azure.core.paging.ItemPaged[ConfigurationSetting]
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:742]:
```suggestion
        :rtype: ~azure.core.paging.ItemPaged[~azure.appconfiguration.ConfigurationSetting]
```

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py [hunk 26]
```diff
@@ -126,19 +133,20 @@ def list_configuration_settings(
...
         :type key_filter: str
         :param label_filter: filter results based on their label. '*' can be
-         used as wildcard in the beginning or end of the filter
+            used as wildcard in the beginning or end of the filter
         :type label_filter: str
         :keyword str accept_datetime: retrieve ConfigurationSetting existed at this datetime
         :keyword list[str] fields: specify which fields to include in the results. Leave None to include all fields
-        :return: An iterator of :class:`ConfigurationSetting`
+        :return: An iterator of :class:`~azure.appconfiguration.ConfigurationSetting`
         :rtype: ~azure.core.async_paging.AsyncItemPaged[ConfigurationSetting]
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py:147]:
```suggestion
        :rtype: ~azure.core.async_paging.AsyncItemPaged[~azure.appconfiguration.ConfigurationSetting]
```

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py [hunk 27]
```diff
@@ -419,19 +440,20 @@ def list_revisions(
...
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py:454]:
```suggestion
        :rtype: ~azure.core.async_paging.AsyncItemPaged[~azure.appconfiguration.ConfigurationSetting]
```

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py [hunk 28]
```diff
@@ -539,6 +563,187 @@ async def set_read_only(
...
+        self, name: str, *, accept_datetime: Optional[str] = None, fields: Optional[List[str]] = None, **kwargs
+    ) -> AsyncItemPaged[ConfigurationSetting]:
+        """List the configuration settings stored under a snapshot in the configuration service, optionally filtered by
+        accept_datetime and fields to present in return.
+
+        :param str name: The snapshot name.
+        :keyword str accept_datetime: Filter out ConfigurationSetting created after this datetime.
+        :keyword list[str] fields: Specify which fields to include in the results. Leave None to include all fields.
+        :return: An iterator of :class:`~azure.appconfiguration.ConfigurationSetting`
+        :rtype: ~azure.core.paging.AsyncItemPaged[ConfigurationSetting]
```
Comment (annatisch) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py:731]:
```suggestion
        :rtype: ~azure.core.paging.AsyncItemPaged[~azure.appconfiguration.ConfigurationSetting]
```

**Review by annatisch — APPROVED** [review]

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py (hunk 24 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:696]: Because we want to get the result unconditionally.
